{"total_count": 19, "incomplete_results": false, "items": [{"url": "https://api.github.com/repos/retr0h/gilt/issues/95", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/95/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/95/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/95/events", "html_url": "https://github.com/retr0h/gilt/issues/95", "id": 596501777, "node_id": "MDU6SXNzdWU1OTY1MDE3Nzc=", "number": 95, "title": "gilt 1.2.3 on pypi not represented by git tag", "user": {"login": "dvzrv", "id": 432519, "node_id": "MDQ6VXNlcjQzMjUxOQ==", "avatar_url": "https://avatars2.githubusercontent.com/u/432519?v=4", "gravatar_id": "", "url": "https://api.github.com/users/dvzrv", "html_url": "https://github.com/dvzrv", "followers_url": "https://api.github.com/users/dvzrv/followers", "following_url": "https://api.github.com/users/dvzrv/following{/other_user}", "gists_url": "https://api.github.com/users/dvzrv/gists{/gist_id}", "starred_url": "https://api.github.com/users/dvzrv/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/dvzrv/subscriptions", "organizations_url": "https://api.github.com/users/dvzrv/orgs", "repos_url": "https://api.github.com/users/dvzrv/repos", "events_url": "https://api.github.com/users/dvzrv/events{/privacy}", "received_events_url": "https://api.github.com/users/dvzrv/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 4, "created_at": "2020-04-08T11:19:03Z", "updated_at": "2020-06-01T21:59:44Z", "closed_at": "2020-04-27T18:44:19Z", "author_association": "NONE", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Bug report\r\n\r\n# Gilt details\r\n\r\n```\r\n1.2.3\r\n```\r\n\r\n- Gilt installation method: One of source, pip\r\n\r\n# Desired Behaviour\r\n\r\nThe versions in this repository and on pypi.org are the same.\r\n\r\n# Actual Behaviour (Bug report only)\r\n\r\ngilt 1.2.3 is released on pypi.org but the latest tag in this repository is for 1.2.2", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/84", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/84/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/84/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/84/events", "html_url": "https://github.com/retr0h/gilt/issues/84", "id": 581368693, "node_id": "MDU6SXNzdWU1ODEzNjg2OTM=", "number": 84, "title": "(retr0h): Remove linting once black support docstring", "user": {"login": "todo[bot]", "id": 32347756, "node_id": "MDM6Qm90MzIzNDc3NTY=", "avatar_url": "https://avatars1.githubusercontent.com/in/5534?v=4", "gravatar_id": "", "url": "https://api.github.com/users/todo%5Bbot%5D", "html_url": "https://github.com/apps/todo", "followers_url": "https://api.github.com/users/todo%5Bbot%5D/followers", "following_url": "https://api.github.com/users/todo%5Bbot%5D/following{/other_user}", "gists_url": "https://api.github.com/users/todo%5Bbot%5D/gists{/gist_id}", "starred_url": "https://api.github.com/users/todo%5Bbot%5D/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/todo%5Bbot%5D/subscriptions", "organizations_url": "https://api.github.com/users/todo%5Bbot%5D/orgs", "repos_url": "https://api.github.com/users/todo%5Bbot%5D/repos", "events_url": "https://api.github.com/users/todo%5Bbot%5D/events{/privacy}", "received_events_url": "https://api.github.com/users/todo%5Bbot%5D/received_events", "type": "Bot", "site_admin": false}, "labels": [{"id": 1909706373, "node_id": "MDU6TGFiZWwxOTA5NzA2Mzcz", "url": "https://api.github.com/repos/retr0h/gilt/labels/todo%20:spiral_notepad:", "name": "todo :spiral_notepad:", "color": "00B0D8", "default": false, "description": null}], "state": "closed", "locked": false, "assignee": {"login": "retr0h", "id": 9895, "node_id": "MDQ6VXNlcjk4OTU=", "avatar_url": "https://avatars1.githubusercontent.com/u/9895?v=4", "gravatar_id": "", "url": "https://api.github.com/users/retr0h", "html_url": "https://github.com/retr0h", "followers_url": "https://api.github.com/users/retr0h/followers", "following_url": "https://api.github.com/users/retr0h/following{/other_user}", "gists_url": "https://api.github.com/users/retr0h/gists{/gist_id}", "starred_url": "https://api.github.com/users/retr0h/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/retr0h/subscriptions", "organizations_url": "https://api.github.com/users/retr0h/orgs", "repos_url": "https://api.github.com/users/retr0h/repos", "events_url": "https://api.github.com/users/retr0h/events{/privacy}", "received_events_url": "https://api.github.com/users/retr0h/received_events", "type": "User", "site_admin": false}, "assignees": [{"login": "retr0h", "id": 9895, "node_id": "MDQ6VXNlcjk4OTU=", "avatar_url": "https://avatars1.githubusercontent.com/u/9895?v=4", "gravatar_id": "", "url": "https://api.github.com/users/retr0h", "html_url": "https://github.com/retr0h", "followers_url": "https://api.github.com/users/retr0h/followers", "following_url": "https://api.github.com/users/retr0h/following{/other_user}", "gists_url": "https://api.github.com/users/retr0h/gists{/gist_id}", "starred_url": "https://api.github.com/users/retr0h/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/retr0h/subscriptions", "organizations_url": "https://api.github.com/users/retr0h/orgs", "repos_url": "https://api.github.com/users/retr0h/repos", "events_url": "https://api.github.com/users/retr0h/events{/privacy}", "received_events_url": "https://api.github.com/users/retr0h/received_events", "type": "User", "site_admin": false}], "milestone": null, "comments": 1, "created_at": "2020-03-14T20:55:12Z", "updated_at": "2020-03-16T03:31:14Z", "closed_at": "2020-03-16T03:31:14Z", "author_association": "NONE", "active_lock_reason": null, "body": "https://github.com/retr0h/gilt/blob/4e213efd0903cf9f1cc0ea90aa4f7ecc80d62f1d/tox.ini#L67-L71\n\n---\n\n###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 4e213efd0903cf9f1cc0ea90aa4f7ecc80d62f1d. It&#x27;s been assigned to @retr0h because they committed the code.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/77", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/77/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/77/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/77/events", "html_url": "https://github.com/retr0h/gilt/issues/77", "id": 581298733, "node_id": "MDU6SXNzdWU1ODEyOTg3MzM=", "number": 77, "title": "custom --runslow pytest option leads to failure on test collection", "user": {"login": "dvzrv", "id": 432519, "node_id": "MDQ6VXNlcjQzMjUxOQ==", "avatar_url": "https://avatars2.githubusercontent.com/u/432519?v=4", "gravatar_id": "", "url": "https://api.github.com/users/dvzrv", "html_url": "https://github.com/dvzrv", "followers_url": "https://api.github.com/users/dvzrv/followers", "following_url": "https://api.github.com/users/dvzrv/following{/other_user}", "gists_url": "https://api.github.com/users/dvzrv/gists{/gist_id}", "starred_url": "https://api.github.com/users/dvzrv/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/dvzrv/subscriptions", "organizations_url": "https://api.github.com/users/dvzrv/orgs", "repos_url": "https://api.github.com/users/dvzrv/repos", "events_url": "https://api.github.com/users/dvzrv/events{/privacy}", "received_events_url": "https://api.github.com/users/dvzrv/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2020-03-14T17:33:46Z", "updated_at": "2020-03-14T18:24:14Z", "closed_at": "2020-03-14T18:24:14Z", "author_association": "NONE", "active_lock_reason": null, "body": "When trying to package version 1.2.1 for Arch Linux I ran into an issue during test collection.\r\n\r\n# Issue Type\r\n\r\n- Bug report\r\n\r\n\r\n# Gilt details\r\n\r\npython-gilt 1.2.1\r\n\r\n- Gilt installation method: source\r\n\r\n# Desired Behaviour\r\n\r\n`pytest -v` runs successfully.\r\n\r\nCurrently I have to remove the custom parameter with a patch, as I otherwise can't even start collection of tests:\r\n\r\n```\r\ndiff -ruN a/test/conftest.py c/test/conftest.py\r\n--- a/test/conftest.py\t2017-05-23 00:29:23.000000000 +0200\r\n+++ c/test/conftest.py\t2020-03-14 18:30:31.910055088 +0100\r\n@@ -30,10 +30,6 @@\r\n pytest_plugins = ['helpers_namespace']\r\n \r\n \r\n-def pytest_addoption(parser):\r\n-    parser.addoption(\"--runslow\", action=\"store_true\", help=\"run slow tests\")\r\n-\r\n-\r\n def random_string(l=5):\r\n     return ''.join(random.choice(string.ascii_uppercase) for _ in range(l))\r\n \r\ndiff -ruN a/test/test_git.py c/test/test_git.py\r\n--- a/test/test_git.py\t2017-05-23 00:29:23.000000000 +0200\r\n+++ c/test/test_git.py\t2020-03-14 18:30:11.133597608 +0100\r\n@@ -28,12 +28,6 @@\r\n \r\n from gilt import git\r\n \r\n-slow = pytest.mark.skipif(\r\n-    not pytest.config.getoption(\"--runslow\"),\r\n-    reason=\"need --runslow option to run\")\r\n-\r\n-\r\n-@slow\r\n def test_clone(temp_dir):\r\n     name = 'retr0h.ansible-etcd'\r\n     repo = 'https://github.com/retr0h/ansible-etcd.git'\r\n@@ -44,7 +38,6 @@\r\n     assert os.path.exists(destination)\r\n \r\n \r\n-@slow\r\n def test_extract(temp_dir):\r\n     name = 'retr0h.ansible-etcd'\r\n     repo = 'https://github.com/retr0h/ansible-etcd.git'\r\n@@ -68,7 +61,6 @@\r\n     assert os.path.exists(giltfile)\r\n \r\n \r\n-@slow\r\n def test_overlay(mocker, temp_dir):\r\n     name = 'lorin.openstack-ansible-modules'\r\n     repo = 'https://github.com/lorin/openstack-ansible-modules.git'\r\n@@ -99,7 +91,6 @@\r\n     assert 2 == len(glob.glob('{}/*'.format(os.path.join(dst_dir, 'tests'))))\r\n \r\n \r\n-@slow\r\n def test_overlay_existing_directory(mocker, temp_dir):\r\n     name = 'lorin.openstack-ansible-modules'\r\n     repo = 'https://github.com/lorin/openstack-ansible-modules.git'\r\n@@ -208,7 +199,6 @@\r\n     assert expected == patched_run_command.mock_calls\r\n \r\n \r\n-@slow\r\n def test_has_version(temp_dir):\r\n     name = 'retr0h.ansible-etcd'\r\n     repo = 'https://github.com/retr0h/ansible-etcd.git'\r\n```\r\n\r\n# Actual Behaviour (Bug report only)\r\n\r\nWhen running\r\n\r\n```\r\npython setup.py build\r\npytest -v\r\n```\r\n\r\nit leads to (also when adding the `--runslow` option to the pytest call):\r\n\r\n```\r\nrunning build\r\nrunning build_py\r\ncreating build\r\ncreating build/lib\r\ncreating build/lib/gilt\r\ncopying gilt/util.py -> build/lib/gilt\r\ncopying gilt/shell.py -> build/lib/gilt\r\ncopying gilt/interpolation.py -> build/lib/gilt\r\ncopying gilt/git.py -> build/lib/gilt\r\ncopying gilt/config.py -> build/lib/gilt\r\ncopying gilt/__init__.py -> build/lib/gilt\r\nrunning egg_info\r\nwriting python_gilt.egg-info/PKG-INFO\r\nwriting dependency_links to python_gilt.egg-info/dependency_links.txt\r\nwriting entry points to python_gilt.egg-info/entry_points.txt\r\nwriting requirements to python_gilt.egg-info/requires.txt\r\nwriting top-level names to python_gilt.egg-info/top_level.txt\r\n[pbr] Reusing existing SOURCES.txt\r\n==> Starting check()...\r\n============================= test session starts ==============================\r\nplatform linux -- Python 3.8.2, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python\r\ncachedir: .pytest_cache\r\nrootdir: /build/python-gilt/src/python-gilt-1.2.1, inifile: pytest.ini, testpaths: test/\r\nplugins: mock-2.0.0, helpers-namespace-2019.1.8, cov-2.8.1\r\ncollecting ... collected 37 items / 1 error / 36 selected\r\n\r\n==================================== ERRORS ====================================\r\n______________________ ERROR collecting test/test_git.py _______________________\r\ntest/test_git.py:32: in <module>\r\n    not pytest.config.getoption(\"--runslow\"),\r\nE   AttributeError: module 'pytest' has no attribute 'config'\r\n!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!\r\n=============================== 1 error in 0.29s ===============================\r\n```", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/76", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/76/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/76/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/76/events", "html_url": "https://github.com/retr0h/gilt/issues/76", "id": 581289896, "node_id": "MDU6SXNzdWU1ODEyODk4OTY=", "number": 76, "title": "test_util.py hardcodes the path for ls ", "user": {"login": "dvzrv", "id": 432519, "node_id": "MDQ6VXNlcjQzMjUxOQ==", "avatar_url": "https://avatars2.githubusercontent.com/u/432519?v=4", "gravatar_id": "", "url": "https://api.github.com/users/dvzrv", "html_url": "https://github.com/dvzrv", "followers_url": "https://api.github.com/users/dvzrv/followers", "following_url": "https://api.github.com/users/dvzrv/following{/other_user}", "gists_url": "https://api.github.com/users/dvzrv/gists{/gist_id}", "starred_url": "https://api.github.com/users/dvzrv/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/dvzrv/subscriptions", "organizations_url": "https://api.github.com/users/dvzrv/orgs", "repos_url": "https://api.github.com/users/dvzrv/repos", "events_url": "https://api.github.com/users/dvzrv/events{/privacy}", "received_events_url": "https://api.github.com/users/dvzrv/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2020-03-14T17:10:12Z", "updated_at": "2020-03-14T20:28:21Z", "closed_at": "2020-03-14T20:28:21Z", "author_association": "NONE", "active_lock_reason": null, "body": "Hi! I'm packaging python-gilt for Arch Linux.\r\nWith version 1.2.1 I'm running into issues with tests.\r\n\r\n# Issue Type\r\n\r\n- Bug report\r\n\r\n# Gilt details\r\n\r\ngilt 1.2.1\r\n\r\n- Gilt installation method: source\r\n\r\n# Desired Behaviour\r\n\r\nThe tests in [test_util.py](https://github.com/retr0h/gilt/blob/master/test/test_util.py#L108-L122) assume, that `ls` is always found in `/bin/ls`.\r\nHowever, this is not true for e.g. Arch Linux (`ls` is in `/usr/bin/ls`).\r\n\r\nIt can be made a little more generic with:\r\n\r\n```\r\ndiff -ruN a/test/test_util.py b/test/test_util.py\r\n--- a/test/test_util.py\t2017-05-23 00:29:23.000000000 +0200\r\n+++ b/test/test_util.py\t2020-03-14 17:57:20.716681417 +0100\r\n@@ -107,16 +107,16 @@\r\n \r\n def test_build_sh_cmd_simple_command():\r\n     cmd = util.build_sh_cmd('ls')\r\n-    assert b'/bin/ls' == cmd._path\r\n+    assert b'/bin/ls' in cmd._path\r\n \r\n \r\n def test_build_sh_cmd_command_with_args():\r\n     cmd = util.build_sh_cmd('ls /tmp')\r\n-    assert b'/bin/ls' == cmd._path\r\n+    assert b'/bin/ls' in cmd._path\r\n     assert [b'/tmp'] == cmd._partial_baked_args\r\n \r\n \r\n def test_build_sh_cmd_command_with_cwd(temp_dir):\r\n     cmd = util.build_sh_cmd('ls', cwd=temp_dir)\r\n-    assert b'/bin/ls' == cmd._path\r\n+    assert b'/bin/ls' in cmd._path\r\n     assert temp_dir == cmd._partial_call_args['cwd']\r\n\r\n```\r\n\r\n# Actual Behaviour (Bug report only)\r\n\r\nAfter running `python setup.py build` and `pytest -v` I get:\r\n\r\n```\r\n=================================== FAILURES ===================================\r\n_______________________ test_build_sh_cmd_simple_command _______________________\r\n\r\n    def test_build_sh_cmd_simple_command():\r\n        cmd = util.build_sh_cmd('ls')\r\n>       assert b'/bin/ls' == cmd._path\r\nE       AssertionError: assert b'/bin/ls' == b'/usr/bin/ls'\r\nE         At index 1 diff: b'b' != b'u'\r\nE         Full diff:\r\nE         - b'/bin/ls'\r\nE         + b'/usr/bin/ls'\r\nE         ?   ++++\r\n\r\n/build/python-gilt/src/python-gilt-1.2.1/test/test_util.py:110: AssertionError\r\n_____________________ test_build_sh_cmd_command_with_args ______________________\r\n\r\n    def test_build_sh_cmd_command_with_args():\r\n        cmd = util.build_sh_cmd('ls /tmp')\r\n>       assert b'/bin/ls' == cmd._path\r\nE       AssertionError: assert b'/bin/ls' == b'/usr/bin/ls'\r\nE         At index 1 diff: b'b' != b'u'\r\nE         Full diff:\r\nE         - b'/bin/ls'\r\nE         + b'/usr/bin/ls'\r\nE         ?   ++++\r\n\r\n/build/python-gilt/src/python-gilt-1.2.1/test/test_util.py:115: AssertionError\r\n______________________ test_build_sh_cmd_command_with_cwd ______________________\r\n\r\ntemp_dir = local('/tmp/pytest-of-builduser/pytest-0/test_build_sh_cmd_command_with0/BEHUZ')\r\n\r\n    def test_build_sh_cmd_command_with_cwd(temp_dir):\r\n        cmd = util.build_sh_cmd('ls', cwd=temp_dir)\r\n>       assert b'/bin/ls' == cmd._path\r\nE       AssertionError: assert b'/bin/ls' == b'/usr/bin/ls'\r\nE         At index 1 diff: b'b' != b'u'\r\nE         Full diff:\r\nE         - b'/bin/ls'\r\nE         + b'/usr/bin/ls'\r\nE         ?   ++++\r\n\r\n/build/python-gilt/src/python-gilt-1.2.1/test/test_util.py:121: AssertionError\r\n==================== 3 failed, 35 passed, 9 errors in 7.50s ====================\r\n```", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/75", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/75/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/75/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/75/events", "html_url": "https://github.com/retr0h/gilt/issues/75", "id": 568421207, "node_id": "MDU6SXNzdWU1Njg0MjEyMDc=", "number": 75, "title": "Add Conditional Overlay for a Source directory/file", "user": {"login": "kuwv", "id": 6035515, "node_id": "MDQ6VXNlcjYwMzU1MTU=", "avatar_url": "https://avatars3.githubusercontent.com/u/6035515?v=4", "gravatar_id": "", "url": "https://api.github.com/users/kuwv", "html_url": "https://github.com/kuwv", "followers_url": "https://api.github.com/users/kuwv/followers", "following_url": "https://api.github.com/users/kuwv/following{/other_user}", "gists_url": "https://api.github.com/users/kuwv/gists{/gist_id}", "starred_url": "https://api.github.com/users/kuwv/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/kuwv/subscriptions", "organizations_url": "https://api.github.com/users/kuwv/orgs", "repos_url": "https://api.github.com/users/kuwv/repos", "events_url": "https://api.github.com/users/kuwv/events{/privacy}", "received_events_url": "https://api.github.com/users/kuwv/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2020-02-20T16:30:24Z", "updated_at": "2020-03-14T20:30:17Z", "closed_at": "2020-03-14T20:30:17Z", "author_association": "NONE", "active_lock_reason": null, "body": "# Issue Type\r\nFeature request\r\n\r\n# Desired Behavour\r\n\r\nGilt is a really useful tool when moving code between environment silos. Sometimes environments contain files that others do not. It would be helpful to add the capability to optionally overlay files when they exist and not fail otherwise.\r\n\r\nIf this already exists then an example would be nice.\r\n\r\nThanks", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/74", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/74/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/74/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/74/events", "html_url": "https://github.com/retr0h/gilt/issues/74", "id": 467197880, "node_id": "MDU6SXNzdWU0NjcxOTc4ODA=", "number": 74, "title": "Test logic failures on openSUSE", "user": {"login": "jayvdb", "id": 15092, "node_id": "MDQ6VXNlcjE1MDky", "avatar_url": "https://avatars1.githubusercontent.com/u/15092?v=4", "gravatar_id": "", "url": "https://api.github.com/users/jayvdb", "html_url": "https://github.com/jayvdb", "followers_url": "https://api.github.com/users/jayvdb/followers", "following_url": "https://api.github.com/users/jayvdb/following{/other_user}", "gists_url": "https://api.github.com/users/jayvdb/gists{/gist_id}", "starred_url": "https://api.github.com/users/jayvdb/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/jayvdb/subscriptions", "organizations_url": "https://api.github.com/users/jayvdb/orgs", "repos_url": "https://api.github.com/users/jayvdb/repos", "events_url": "https://api.github.com/users/jayvdb/events{/privacy}", "received_events_url": "https://api.github.com/users/jayvdb/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2019-07-12T03:05:46Z", "updated_at": "2020-03-14T20:30:49Z", "closed_at": "2020-03-14T20:30:49Z", "author_association": "NONE", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Bug report\r\n\r\n# Gilt details\r\n\r\n- Gilt installation method: rpm :P (using pypi release 1.2.1)\r\n\r\n# Desired Behaviour\r\n\r\nTest logic allow for `ls` to also be located in `/usr/bin/ls` where it occurs for openSUSE.\r\n\r\nCurrently it only supports `/bin/ls`.\r\n\r\nThis is part of a push to package gilt for openSUSE\r\n\r\n# Actual Behaviour (Bug report only)\r\n\r\nhttps://build.opensuse.org/package/show/home:jayvdb:py-new/python-python-gilt\r\n```console\r\n[   66s] =================================== FAILURES ===================================\r\n[   66s] _______________________ test_build_sh_cmd_simple_command _______________________\r\n[   66s] \r\n[   66s]     def test_build_sh_cmd_simple_command():\r\n[   66s]         cmd = util.build_sh_cmd('ls')\r\n[   66s] >       assert b'/bin/ls' == cmd._path\r\n[   66s] E       AssertionError: assert '/bin/ls' == '/usr/bin/ls'\r\n[   66s] E         - /bin/ls\r\n[   66s] E         + /usr/bin/ls\r\n[   66s] E         ? ++++\r\n[   66s] \r\n[   66s] /home/abuild/rpmbuild/BUILD/python-gilt-1.2.1/test/test_util.py:110: AssertionError\r\n[   66s] _____________________ test_build_sh_cmd_command_with_args ______________________\r\n[   66s] \r\n[   66s]     def test_build_sh_cmd_command_with_args():\r\n[   66s]         cmd = util.build_sh_cmd('ls /tmp')\r\n[   66s] >       assert b'/bin/ls' == cmd._path\r\n[   66s] E       AssertionError: assert '/bin/ls' == '/usr/bin/ls'\r\n[   66s] E         - /bin/ls\r\n[   66s] E         + /usr/bin/ls\r\n[   66s] E         ? ++++\r\n[   66s] \r\n[   66s] /home/abuild/rpmbuild/BUILD/python-gilt-1.2.1/test/test_util.py:115: AssertionError\r\n[   66s] ______________________ test_build_sh_cmd_command_with_cwd ______________________\r\n[   66s] \r\n[   66s] temp_dir = local('/tmp/pytest-of-abuild/pytest-0/test_build_sh_cmd_command_with0/LHUER')\r\n[   66s] \r\n[   66s]     def test_build_sh_cmd_command_with_cwd(temp_dir):\r\n[   66s]         cmd = util.build_sh_cmd('ls', cwd=temp_dir)\r\n[   66s] >       assert b'/bin/ls' == cmd._path\r\n[   66s] E       AssertionError: assert '/bin/ls' == '/usr/bin/ls'\r\n[   66s] E         - /bin/ls\r\n[   66s] E         + /usr/bin/ls\r\n[   66s] E         ? ++++\r\n[   66s] \r\n[   66s] /home/abuild/rpmbuild/BUILD/python-gilt-1.2.1/test/test_util.py:121: AssertionError\r\n```\r\n", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/72", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/72/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/72/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/72/events", "html_url": "https://github.com/retr0h/gilt/issues/72", "id": 439023083, "node_id": "MDU6SXNzdWU0MzkwMjMwODM=", "number": 72, "title": "git-url-parse doesn't properly parse Bitbucket Server URIs", "user": {"login": "philipsd6", "id": 908201, "node_id": "MDQ6VXNlcjkwODIwMQ==", "avatar_url": "https://avatars1.githubusercontent.com/u/908201?v=4", "gravatar_id": "", "url": "https://api.github.com/users/philipsd6", "html_url": "https://github.com/philipsd6", "followers_url": "https://api.github.com/users/philipsd6/followers", "following_url": "https://api.github.com/users/philipsd6/following{/other_user}", "gists_url": "https://api.github.com/users/philipsd6/gists{/gist_id}", "starred_url": "https://api.github.com/users/philipsd6/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/philipsd6/subscriptions", "organizations_url": "https://api.github.com/users/philipsd6/orgs", "repos_url": "https://api.github.com/users/philipsd6/repos", "events_url": "https://api.github.com/users/philipsd6/events{/privacy}", "received_events_url": "https://api.github.com/users/philipsd6/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2019-04-30T23:45:57Z", "updated_at": "2020-03-14T20:39:02Z", "closed_at": "2020-03-14T20:39:02Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "\r\n  # Issue Type\r\n\r\n  - Bug report\r\n\r\n  # Gilt details\r\n\r\n  ```\r\n  gilt, version 1.2.2.dev2\r\n  ```\r\n\r\n  - Gilt installation method: source\r\n\r\n  # Desired Behaviour\r\n\r\n  Running `gilt overlay` using this `gilt.yml` config file:\r\n  ```yaml\r\n  - git: https://our.bitbucket.server/git/scm/myproject/my-helloworld-roles.git\r\n    version: master\r\n    files:\r\n      - src: helloworld_*\r\n        dst: roles/\r\n  ```\r\n  I expect everything matching `helloworld_.*` to be put in `./roles`\r\n\r\n  # Actual Behaviour (Bug report only)\r\n\r\n  ```pytb\r\n  $ gilt --debug overlay\r\n  Traceback (most recent call last):\r\n    File \"/home/philipsd6/.local/bin/gilt\", line 10, in <module>\r\n      sys.exit(main())\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/click/core.py\", line 722, in __call__\r\n      return self.main(*args, **kwargs)\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/click/core.py\", line 697, in main\r\n      rv = self.invoke(ctx)\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/click/core.py\", line 1066, in invoke\r\n      return _process_result(sub_ctx.command.invoke(sub_ctx))\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/click/core.py\", line 895, in invoke\r\n      return ctx.invoke(self.callback, **ctx.params)\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/click/core.py\", line 535, in invoke\r\n      return callback(*args, **kwargs)\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/click/decorators.py\", line 17, in new_func\r\n      return f(get_current_context(), *args, **kwargs)\r\n    File \"/home/philipsd6/devel/gilt/gilt/shell.py\", line 68, in overlay\r\n      with fasteners.InterProcessLock(c.lock_file):\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/fasteners/process_lock.py\", line 174, in __enter__\r\n      self.acquire()\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/fasteners/process_lock.py\", line 151, in acquire\r\n      self._do_open()\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/fasteners/process_lock.py\", line 115, in _do_open\r\n      made_basedir = _ensure_tree(basedir)\r\n    File \"/home/philipsd6/.local/lib/python3.6/site-packages/fasteners/process_lock.py\", line 37, in _ensure_tree\r\n      os.makedirs(path)\r\n    File \"/usr/lib/python3.6/os.py\", line 210, in makedirs\r\n      makedirs(head, mode, exist_ok)\r\n    File \"/usr/lib/python3.6/os.py\", line 210, in makedirs\r\n      makedirs(head, mode, exist_ok)\r\n    File \"/usr/lib/python3.6/os.py\", line 220, in makedirs\r\n      mkdir(name, mode)\r\n  PermissionError: [Errno 13] Permission denied: '/our.bitbucket.server'\r\n  ```\r\n\r\nThe problem is in the git-url-parse dependent package:\r\n```pycon\r\nIn [1]: from giturlparse import parse\r\nIn [2]: p = parse('https://our.bitbucket.server/git/scm/myproject/my-helloworld-roles.git')\r\nIn [3]: assert p.name == 'my-helloworld-roles'\r\nIn [4]: assert p.owner == 'myproject', f\"p.owner is incorrect: {p.owner}\"\r\n---------------------------------------------------------------------------\r\nAssertionError                            Traceback (most recent call last)\r\n<ipython-input-4-a20d961cd7d8> in <module>\r\n----> 1 assert p.owner == 'myproject', f\"p.owner is incorrect: {p.owner}\"\r\n\r\nAssertionError: p.owner is incorrect: /our.bitbucket.server/git/scm/myproject\r\n\r\nIn [5]: assert p.resource == 'our.bitbucket.server', f\"p.resource is incorrect: {p.resource}\"\r\n---------------------------------------------------------------------------\r\nAssertionError                            Traceback (most recent call last)\r\n<ipython-input-5-88ee6374fc22> in <module>\r\n----> 1 assert p.resource == 'our.bitbucket.server', f\"p.resource is incorrect: {p.resource}\"\r\n\r\nAssertionError: p.resource is incorrect: https\r\n\r\nIn [6]: assert p.protocol == 'https', f\"p.protocol is incorrect: {p.protocol}\"\r\n---------------------------------------------------------------------------\r\nAssertionError                            Traceback (most recent call last)\r\n<ipython-input-12-86dec67945be> in <module>\r\n----> 1 assert p.protocol == 'https', f\"p.protocol is incorrect: {p.protocol}\"\r\n\r\nAssertionError: p.protocol is incorrect: ssh\r\n```", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/69", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/69/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/69/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/69/events", "html_url": "https://github.com/retr0h/gilt/issues/69", "id": 417201210, "node_id": "MDU6SXNzdWU0MTcyMDEyMTA=", "number": 69, "title": "simplify tox environments", "user": {"login": "ssbarnea", "id": 102495, "node_id": "MDQ6VXNlcjEwMjQ5NQ==", "avatar_url": "https://avatars3.githubusercontent.com/u/102495?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ssbarnea", "html_url": "https://github.com/ssbarnea", "followers_url": "https://api.github.com/users/ssbarnea/followers", "following_url": "https://api.github.com/users/ssbarnea/following{/other_user}", "gists_url": "https://api.github.com/users/ssbarnea/gists{/gist_id}", "starred_url": "https://api.github.com/users/ssbarnea/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ssbarnea/subscriptions", "organizations_url": "https://api.github.com/users/ssbarnea/orgs", "repos_url": "https://api.github.com/users/ssbarnea/repos", "events_url": "https://api.github.com/users/ssbarnea/events{/privacy}", "received_events_url": "https://api.github.com/users/ssbarnea/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2019-03-05T09:34:28Z", "updated_at": "2020-03-16T04:01:22Z", "closed_at": "2020-03-16T04:01:22Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Feature request\r\n\r\n- Gilt installation method: One of source, pip\r\n\r\n# Actual Behaviour (Bug report only)\r\n\r\nCurrent list of tox targes does not provide an optimal list of tox environment for testing the code:\r\n```\r\n$ tox -a                                                                                                                                                                                         py27-unit\r\npy36-unit\r\npy27-lint\r\npy36-lint\r\ndoc\r\nformat\r\n```\r\n\r\n* no need to lint with two versions of python, using  one with basepython=python3 is enough, it is key to run lint with newer python3 for the better coverage but is not needed to run lint on older pythons.\r\n* format should be considered a linter and included in generic 'lint' environment\r\n* format run should be implicit when running tox to avoid regressions\r\n* it would be easier to have simplified unittest envrionments like just \"py27\", easier to type an extend\r\n* we should enable other interpreters py35,py37 for start\r\n* we should enable skip_missing_interpreters in order to ease development testing on machines that may not have a full range on interpreters (CI should run all)\r\n\r\n# Desired Behaviour\r\n```\r\n$ tox -a\r\nlint\r\npy27\r\npy37\r\npy36\r\npy35\r\ndoc\r\n```\r\n\r\nRunning just `tox` on a machine that does not have one of listed interpreters will not trigger an error.\r\n\r\nNote: I can easily make PR that implements this but before doing it I will wait for some feedback in order to see if it desired.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/68", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/68/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/68/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/68/events", "html_url": "https://github.com/retr0h/gilt/issues/68", "id": 417195387, "node_id": "MDU6SXNzdWU0MTcxOTUzODc=", "number": 68, "title": "gilt needs a ci", "user": {"login": "ssbarnea", "id": 102495, "node_id": "MDQ6VXNlcjEwMjQ5NQ==", "avatar_url": "https://avatars3.githubusercontent.com/u/102495?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ssbarnea", "html_url": "https://github.com/ssbarnea", "followers_url": "https://api.github.com/users/ssbarnea/followers", "following_url": "https://api.github.com/users/ssbarnea/following{/other_user}", "gists_url": "https://api.github.com/users/ssbarnea/gists{/gist_id}", "starred_url": "https://api.github.com/users/ssbarnea/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ssbarnea/subscriptions", "organizations_url": "https://api.github.com/users/ssbarnea/orgs", "repos_url": "https://api.github.com/users/ssbarnea/repos", "events_url": "https://api.github.com/users/ssbarnea/events{/privacy}", "received_events_url": "https://api.github.com/users/ssbarnea/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2019-03-05T09:20:42Z", "updated_at": "2019-03-05T18:03:44Z", "closed_at": "2019-03-05T18:03:44Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Feature request\r\n\r\n# Gilt details\r\n\r\nGilt project is great and it needs a CI for testing any contributions made to it. It does not matter what CI is using and where is running as long it would post results on raised PRs. so we would be able to see what happens with proposed changed.\r\n\r\nTesting changes locally with tox is good, but the results would be greatly influenced by the local machine of each developer, and this means unpredictable results and more bugs getting in.\r\n\r\nPlease consider this.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/66", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/66/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/66/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/66/events", "html_url": "https://github.com/retr0h/gilt/issues/66", "id": 416915644, "node_id": "MDU6SXNzdWU0MTY5MTU2NDQ=", "number": 66, "title": "tox unit tests are broken when GILT_CACHE_DIRECTORY is defined", "user": {"login": "ssbarnea", "id": 102495, "node_id": "MDQ6VXNlcjEwMjQ5NQ==", "avatar_url": "https://avatars3.githubusercontent.com/u/102495?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ssbarnea", "html_url": "https://github.com/ssbarnea", "followers_url": "https://api.github.com/users/ssbarnea/followers", "following_url": "https://api.github.com/users/ssbarnea/following{/other_user}", "gists_url": "https://api.github.com/users/ssbarnea/gists{/gist_id}", "starred_url": "https://api.github.com/users/ssbarnea/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ssbarnea/subscriptions", "organizations_url": "https://api.github.com/users/ssbarnea/orgs", "repos_url": "https://api.github.com/users/ssbarnea/repos", "events_url": "https://api.github.com/users/ssbarnea/events{/privacy}", "received_events_url": "https://api.github.com/users/ssbarnea/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2019-03-04T17:40:23Z", "updated_at": "2020-03-14T17:51:21Z", "closed_at": "2019-03-05T18:00:42Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Bug report\r\n\r\n# Actual Behaviour (Bug report only)\r\n\r\n```\r\n$ tox                                                                                                                                                                                           [17:37:20]\r\nGLOB sdist-make: /Users/ssbarnea/os/gilt/setup.py\r\npy27-unit inst-nodeps: /Users/ssbarnea/os/gilt/.tox/.tmp/package/1/python-gilt-1.2.2.dev1.zip\r\npy27-unit installed: DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.,atomicwrites==1.3.0,attrs==19.1.0,Click==7.0,colorama==0.4.1,configparser==3.7.3,coverage==4.5.2,entrypoints==0.3,enum34==1.1.6,fasteners==0.14.1,flake8==3.7.7,funcsigs==1.0.2,functools32==3.2.3.post2,git-url-parse==1.2.0,mccabe==0.6.1,mock==2.0.0,monotonic==1.5,more-itertools==5.0.0,pathlib2==2.3.3,pbr==5.1.3,pluggy==0.9.0,py==1.8.0,pycodestyle==2.5.0,pyflakes==2.1.1,pytest==4.3.0,pytest-cov==2.6.1,pytest-helpers-namespace==2019.1.8,pytest-mock==1.10.1,python-gilt==1.2.2.dev1,PyYAML==3.13,scandir==1.9.0,sh==1.12.14,six==1.12.0,typing==3.6.6,yapf==0.16.3\r\npy27-unit run-test-pre: PYTHONHASHSEED='3184049481'\r\npy27-unit runtests: commands[0] | py.test --runslow -vv --cov-report=term-missing --cov=/Users/ssbarnea/os/gilt/gilt/ --no-cov-on-fail\r\n=========================================================================================== test session starts ===========================================================================================\r\nplatform darwin -- Python 2.7.15, pytest-4.3.0, py-1.8.0, pluggy-0.9.0 -- /Users/ssbarnea/os/gilt/.tox/py27-unit/bin/python\r\ncachedir: .tox/py27-unit/.pytest_cache\r\nrootdir: /Users/ssbarnea/os/gilt, inifile: pytest.ini\r\nplugins: mock-1.10.1, helpers-namespace-2019.1.8, cov-2.6.1\r\ncollected 47 items\r\n\r\ntest/test_config.py::test_config[gilt_data] ERROR                                                                                                                                                   [  2%]\r\ntest/test_config.py::test_config_missing_git_key[missing_git_key_data] ERROR                                                                                                                        [  4%]\r\ntest/test_config.py::test_config_missing_version_key[missing_version_key_data] ERROR                                                                                                                [  6%]\r\ntest/test_config.py::test_config_missing_dst_key[missing_dst_key_data] ERROR                                                                                                                        [  8%]\r\ntest/test_config.py::test_config_missing_files_src_key[missing_files_src_key_data] ERROR                                                                                                            [ 10%]\r\ntest/test_config.py::test_config_missing_files_dst_key[missing_files_dst_key_data] ERROR                                                                                                            [ 12%]\r\ntest/test_config.py::test_get_config_generator[gilt_data] ERROR                                                                                                                                     [ 14%]\r\ntest/test_config.py::test_get_files_generator PASSED                                                                                                                                                [ 17%]\r\ntest/test_config.py::test_get_config[gilt_data] ERROR                                                                                                                                               [ 19%]\r\ntest/test_config.py::test_get_config_handles_parse_error[invalid_gilt_data] ERROR                                                                                                                   [ 21%]\r\ntest/test_config.py::test_get_dst_dir PASSED                                                                                                                                                        [ 23%]\r\ntest/test_config.py::test_get_clone_dir FAILED                                                                                                                                                      [ 25%]\r\ntest/test_config.py::test_makedirs PASSED\r\n```\r\n\r\n\r\nMainly there are 7-8 tests that are failing on any platform, which prevents any fix from being proposed so far.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/64", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/64/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/64/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/64/events", "html_url": "https://github.com/retr0h/gilt/issues/64", "id": 366451653, "node_id": "MDU6SXNzdWUzNjY0NTE2NTM=", "number": 64, "title": "Add support for specifying clone depth (shallow clones)", "user": {"login": "bootswithdefer", "id": 415790, "node_id": "MDQ6VXNlcjQxNTc5MA==", "avatar_url": "https://avatars0.githubusercontent.com/u/415790?v=4", "gravatar_id": "", "url": "https://api.github.com/users/bootswithdefer", "html_url": "https://github.com/bootswithdefer", "followers_url": "https://api.github.com/users/bootswithdefer/followers", "following_url": "https://api.github.com/users/bootswithdefer/following{/other_user}", "gists_url": "https://api.github.com/users/bootswithdefer/gists{/gist_id}", "starred_url": "https://api.github.com/users/bootswithdefer/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/bootswithdefer/subscriptions", "organizations_url": "https://api.github.com/users/bootswithdefer/orgs", "repos_url": "https://api.github.com/users/bootswithdefer/repos", "events_url": "https://api.github.com/users/bootswithdefer/events{/privacy}", "received_events_url": "https://api.github.com/users/bootswithdefer/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 478592829, "node_id": "MDU6TGFiZWw0Nzg1OTI4Mjk=", "url": "https://api.github.com/repos/retr0h/gilt/labels/invalid", "name": "invalid", "color": "e6e6e6", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2018-10-03T17:42:14Z", "updated_at": "2019-06-18T08:11:11Z", "closed_at": "2018-10-04T15:57:31Z", "author_association": "NONE", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Feature request\r\n\r\n# Gilt details\r\n\r\n```\r\ngilt, version 1.2.1\r\n```\r\n\r\n- Gilt installation method: pip\r\n\r\n# Desired Behaviour\r\n\r\nAllow specifying clone depth (shallow clones) with git.  This will make large/old repos clone faster and use less disk space when you just want the head of master.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/62", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/62/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/62/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/62/events", "html_url": "https://github.com/retr0h/gilt/issues/62", "id": 323196325, "node_id": "MDU6SXNzdWUzMjMxOTYzMjU=", "number": 62, "title": "gilt throwing rather useless error... ", "user": {"login": "sandinak", "id": 1916398, "node_id": "MDQ6VXNlcjE5MTYzOTg=", "avatar_url": "https://avatars1.githubusercontent.com/u/1916398?v=4", "gravatar_id": "", "url": "https://api.github.com/users/sandinak", "html_url": "https://github.com/sandinak", "followers_url": "https://api.github.com/users/sandinak/followers", "following_url": "https://api.github.com/users/sandinak/following{/other_user}", "gists_url": "https://api.github.com/users/sandinak/gists{/gist_id}", "starred_url": "https://api.github.com/users/sandinak/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/sandinak/subscriptions", "organizations_url": "https://api.github.com/users/sandinak/orgs", "repos_url": "https://api.github.com/users/sandinak/repos", "events_url": "https://api.github.com/users/sandinak/events{/privacy}", "received_events_url": "https://api.github.com/users/sandinak/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 478592829, "node_id": "MDU6TGFiZWw0Nzg1OTI4Mjk=", "url": "https://api.github.com/repos/retr0h/gilt/labels/invalid", "name": "invalid", "color": "e6e6e6", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2018-05-15T12:30:03Z", "updated_at": "2018-10-04T15:53:21Z", "closed_at": "2018-10-04T15:53:14Z", "author_association": "NONE", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Bug report\r\n\r\n# Gilt details\r\n\r\n```\r\ngilt, version **1.2.1**\r\n```\r\n\r\n- Gilt installation method:  pip\r\n\r\n# Desired Behaviour\r\n\r\ngilting in repo\r\n\r\n# Actual Behaviour (Bug report only)\r\n\r\n```\r\n> gilt --debug overlay\r\nTraceback (most recent call last):\r\n  File \"/Users/brmathes/git/ansible-mc/venv/bin/gilt\", line 11, in <module>\r\n    sys.exit(main())\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/click/core.py\", line 722, in __call__\r\n    return self.main(*args, **kwargs)\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/click/core.py\", line 697, in main\r\n    rv = self.invoke(ctx)\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/click/core.py\", line 1066, in invoke\r\n    return _process_result(sub_ctx.command.invoke(sub_ctx))\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/click/core.py\", line 895, in invoke\r\n    return ctx.invoke(self.callback, **ctx.params)\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/click/core.py\", line 535, in invoke\r\n    return callback(*args, **kwargs)\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/click/decorators.py\", line 17, in new_func\r\n    return f(get_current_context(), *args, **kwargs)\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/gilt/shell.py\", line 67, in overlay\r\n    for c in config.config(filename):\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/gilt/config.py\", line 60, in config\r\n    return [Config(**d) for d in _get_config_generator(filename)]\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/gilt/config.py\", line 96, in _get_config_generator\r\n    dst_dir = _get_dst_dir(d['dst'])\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/gilt/config.py\", line 155, in _get_dst_dir\r\n    _makedirs(dst_dir)\r\n  File \"/Users/brmathes/git/ansible-mc/venv/lib/python2.7/site-packages/gilt/config.py\", line 205, in _makedirs\r\n    os.makedirs(dirname)\r\n  File \"/Users/brmathes/git/ansible-mc/venv/bin/../lib/python2.7/os.py\", line 157, in makedirs\r\n    mkdir(name, mode)\r\n```\r\nSo the issue here:\r\n - dst: must have a / if it's a directory .. shouldnt' have to have that.  The relevant code is:\r\n```\r\n  dirname, _ = os.path.split(path)\r\n    try:\r\n        os.makedirs(dirname)\r\n    except OSError as exc:\r\n        if exc.errno == errno.EEXIST:\r\n            pass\r\n        else:\r\n            raise\r\n```\r\ndirname will only be set if sees more than one thing in the path. \r\n", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/61", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/61/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/61/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/61/events", "html_url": "https://github.com/retr0h/gilt/issues/61", "id": 303868521, "node_id": "MDU6SXNzdWUzMDM4Njg1MjE=", "number": 61, "title": "Feature to allow full clone?", "user": {"login": "beddari", "id": 692371, "node_id": "MDQ6VXNlcjY5MjM3MQ==", "avatar_url": "https://avatars3.githubusercontent.com/u/692371?v=4", "gravatar_id": "", "url": "https://api.github.com/users/beddari", "html_url": "https://github.com/beddari", "followers_url": "https://api.github.com/users/beddari/followers", "following_url": "https://api.github.com/users/beddari/following{/other_user}", "gists_url": "https://api.github.com/users/beddari/gists{/gist_id}", "starred_url": "https://api.github.com/users/beddari/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/beddari/subscriptions", "organizations_url": "https://api.github.com/users/beddari/orgs", "repos_url": "https://api.github.com/users/beddari/repos", "events_url": "https://api.github.com/users/beddari/events{/privacy}", "received_events_url": "https://api.github.com/users/beddari/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2018-03-09T14:58:04Z", "updated_at": "2018-03-10T16:28:14Z", "closed_at": "2018-03-10T16:28:14Z", "author_association": "NONE", "active_lock_reason": null, "body": "Hi there,\r\nwondering, would you take a new feature adding \"clone\" to the gilt.yml format? Right now we're hacking around our need by setting GIT_DIR manually to the cached gilt cloned repo.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/60", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/60/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/60/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/60/events", "html_url": "https://github.com/retr0h/gilt/issues/60", "id": 264011876, "node_id": "MDU6SXNzdWUyNjQwMTE4NzY=", "number": 60, "title": "Add command to clear the cache for a repo", "user": {"login": "chrisplo", "id": 13784493, "node_id": "MDQ6VXNlcjEzNzg0NDkz", "avatar_url": "https://avatars2.githubusercontent.com/u/13784493?v=4", "gravatar_id": "", "url": "https://api.github.com/users/chrisplo", "html_url": "https://github.com/chrisplo", "followers_url": "https://api.github.com/users/chrisplo/followers", "following_url": "https://api.github.com/users/chrisplo/following{/other_user}", "gists_url": "https://api.github.com/users/chrisplo/gists{/gist_id}", "starred_url": "https://api.github.com/users/chrisplo/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/chrisplo/subscriptions", "organizations_url": "https://api.github.com/users/chrisplo/orgs", "repos_url": "https://api.github.com/users/chrisplo/repos", "events_url": "https://api.github.com/users/chrisplo/events{/privacy}", "received_events_url": "https://api.github.com/users/chrisplo/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2017-10-09T20:10:32Z", "updated_at": "2020-03-14T20:27:57Z", "closed_at": "2020-03-14T20:27:57Z", "author_association": "NONE", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Bug report\r\n\r\n# Gilt details\r\n\r\ngilt, version 1.1.0, via pip\r\n\r\n# Desired Behaviour\r\nRun `git remote prune origin` and retry if there is an issue\r\n\r\n# Actual Behaviour\r\n\r\nWhile gilt is running `/usr/bin/git pull --rebase --ff-only`\r\n```\r\nerror: unable to resolve reference refs/remotes/origin/4.6.0/hotfix: Not a directory\r\nFrom github.com:metacloud/ansible-systems\r\n ! [new branch]      4.6.0/hotfix -> origin/4.6.0/hotfix  (unable to update local ref)\r\n * [new tag]         4.6.0      -> 4.6.0\r\nerror: some local refs could not be updated; try running\r\n 'git remote prune origin' to remove any old, conflicting branches\r\n```\r\n", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/41", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/41/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/41/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/41/events", "html_url": "https://github.com/retr0h/gilt/issues/41", "id": 231491608, "node_id": "MDU6SXNzdWUyMzE0OTE2MDg=", "number": 41, "title": "Test Failure: test_makedirs fails on Ubuntu Linux 16.04", "user": {"login": "ephur", "id": 926696, "node_id": "MDQ6VXNlcjkyNjY5Ng==", "avatar_url": "https://avatars2.githubusercontent.com/u/926696?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ephur", "html_url": "https://github.com/ephur", "followers_url": "https://api.github.com/users/ephur/followers", "following_url": "https://api.github.com/users/ephur/following{/other_user}", "gists_url": "https://api.github.com/users/ephur/gists{/gist_id}", "starred_url": "https://api.github.com/users/ephur/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ephur/subscriptions", "organizations_url": "https://api.github.com/users/ephur/orgs", "repos_url": "https://api.github.com/users/ephur/repos", "events_url": "https://api.github.com/users/ephur/events{/privacy}", "received_events_url": "https://api.github.com/users/ephur/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 478592827, "node_id": "MDU6TGFiZWw0Nzg1OTI4Mjc=", "url": "https://api.github.com/repos/retr0h/gilt/labels/enhancement", "name": "enhancement", "color": "84b6eb", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 3, "created_at": "2017-05-25T23:59:42Z", "updated_at": "2017-06-12T22:42:30Z", "closed_at": "2017-06-12T22:13:57Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Bug report\r\n- Feature request\r\n\r\n# Gilt details\r\n\r\n```\r\ngilt 1.1.0    /  master\r\n```\r\n\r\n# Desired Behaviour\r\nTests should pass when running tox\r\n\r\n# Actual Behaviour (Bug report only)\r\ntest/test_config.py::test_makedirs FAILED\r\n```\r\n========================================================================== FAILURES ==========================================================================\r\n_______________________________________________________________________ test_makedirs ________________________________________________________________________\r\n\r\ntemp_dir = local('/tmp/pytest-of-ephur/pytest-3/test_makedirs0/SLXGL')\r\n\r\n    def test_makedirs(temp_dir):\r\n        config._makedirs('foo/')\r\n    \r\n        d = os.path.join(temp_dir.strpath, 'foo')\r\n        assert os.path.isdir(d)\r\n    \r\n        expected = (7 * 64 + 5 * 8 + 5)  # Octal 755\r\n>       assert expected == (os.lstat(d).st_mode & 0o777)\r\nE       AssertionError: assert 493 == (16893 & 511)\r\nE        +  where 16893 = posix.stat_result(st_mode=16893, st_ino=15337763, st_dev=2067, st_nlink=2, st_...0, st_size=4096, st_atime=1495756545, st_mtime=1495\r\n756545, st_ctime=1495756545).st_mode\r\nE        +    where posix.stat_result(st_mode=16893, st_ino=15337763, st_dev=2067, st_nlink=2, st_...0, st_size=4096, st_atime=1495756545, st_mtime=1495756545\r\n, st_ctime=1495756545) = <built-in function lstat>('/tmp/pytest-of-ephur/pytest-3/test_makedirs0/SLXGL/foo')\r\nE        +      where <built-in function lstat> = os.lstat\r\n\r\n/home/ephur/Projects/or/upstream_gilt/gilt/test/test_config.py:202: AssertionError\r\n```\r\n\r\nThis happens on ubuntu / python 2.7.12. Test passes okay on OSX.\r\n```\r\n \u2718 \ue0c8 (or) \ue0c8 ephur@earthly \ue0c8 ~/Projects/or/upstream_gilt/gilt \ue0c8 \u00b1 master  \ue0c8 python --version\r\nPython 2.7.12\r\n (or) \ue0c8 ephur@earthly \ue0c8 ~/Projects/or/upstream_gilt/gilt \ue0c8 \u00b1 master  \ue0c8 cat /etc/lsb-release\r\nDISTRIB_ID=Ubuntu\r\nDISTRIB_RELEASE=16.04\r\nDISTRIB_CODENAME=xenial\r\nDISTRIB_DESCRIPTION=\"Ubuntu 16.04.2 LTS\"\r\n (or) \ue0c8 ephur@earthly \ue0c8 ~/Projects/or/upstream_gilt/gilt \ue0c8 \u00b1 master  \ue0c8 uname -r \r\n4.8.0-53-generic\r\n (or) \ue0c8 ephur@earthly \ue0c8 ~/Projects/or/upstream_gilt/gilt \ue0c8 \u00b1 master  \ue0c8\r\n```\r\n", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/40", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/40/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/40/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/40/events", "html_url": "https://github.com/retr0h/gilt/issues/40", "id": 231412349, "node_id": "MDU6SXNzdWUyMzE0MTIzNDk=", "number": 40, "title": "Feature: Merge Directories instead of Overwrite", "user": {"login": "ephur", "id": 926696, "node_id": "MDQ6VXNlcjkyNjY5Ng==", "avatar_url": "https://avatars2.githubusercontent.com/u/926696?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ephur", "html_url": "https://github.com/ephur", "followers_url": "https://api.github.com/users/ephur/followers", "following_url": "https://api.github.com/users/ephur/following{/other_user}", "gists_url": "https://api.github.com/users/ephur/gists{/gist_id}", "starred_url": "https://api.github.com/users/ephur/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ephur/subscriptions", "organizations_url": "https://api.github.com/users/ephur/orgs", "repos_url": "https://api.github.com/users/ephur/repos", "events_url": "https://api.github.com/users/ephur/events{/privacy}", "received_events_url": "https://api.github.com/users/ephur/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 478592827, "node_id": "MDU6TGFiZWw0Nzg1OTI4Mjc=", "url": "https://api.github.com/repos/retr0h/gilt/labels/enhancement", "name": "enhancement", "color": "84b6eb", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2017-05-25T17:58:27Z", "updated_at": "2017-12-12T16:42:43Z", "closed_at": "2017-12-12T16:42:42Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "# Issue Type\r\n- Feature request\r\n\r\n# Gilt details\r\n\r\n```\r\n1.1.1.dev6 ..... \r\n```\r\n\r\n# Desired Behaviour\r\n\r\nIn using gilt to manage terraform environments, there are some files that will remain in a directory that gets files from another repo put on top of them. The current shutil.copytree seems to enforce removing the directory first. I believe that a merge should be possible instead of an overwrite.\r\n\r\nWould this feature be worth adding in general?\r\n\r\nI am using it locally,  not sure if this is really the best way to implement it (and certainly would want it to default to off if it were merged upstream to keep default behavior the same). \r\n\r\nhttps://github.com/metacloud/gilt/compare/master...ephur:master", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/38", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/38/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/38/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/38/events", "html_url": "https://github.com/retr0h/gilt/issues/38", "id": 230433251, "node_id": "MDU6SXNzdWUyMzA0MzMyNTE=", "number": 38, "title": "FEATURE: Ability to read/reference environment variables in gilt.yml", "user": {"login": "ephur", "id": 926696, "node_id": "MDQ6VXNlcjkyNjY5Ng==", "avatar_url": "https://avatars2.githubusercontent.com/u/926696?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ephur", "html_url": "https://github.com/ephur", "followers_url": "https://api.github.com/users/ephur/followers", "following_url": "https://api.github.com/users/ephur/following{/other_user}", "gists_url": "https://api.github.com/users/ephur/gists{/gist_id}", "starred_url": "https://api.github.com/users/ephur/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ephur/subscriptions", "organizations_url": "https://api.github.com/users/ephur/orgs", "repos_url": "https://api.github.com/users/ephur/repos", "events_url": "https://api.github.com/users/ephur/events{/privacy}", "received_events_url": "https://api.github.com/users/ephur/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 478592827, "node_id": "MDU6TGFiZWw0Nzg1OTI4Mjc=", "url": "https://api.github.com/repos/retr0h/gilt/labels/enhancement", "name": "enhancement", "color": "84b6eb", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2017-05-22T15:28:06Z", "updated_at": "2017-06-12T22:42:27Z", "closed_at": "2017-05-23T02:04:41Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "# Issue Type\r\n- Feature request\r\n\r\n# Gilt details\r\n1.1.0 from pip\r\n\r\n# Desired Behaviour\r\nI'm working on doing a build with Terraform and we use environment variables to set a lot of values in our CI testing. It would be useful to support environment variables in the yml parsing, for example\r\n\r\n- git: https://github.com/retr0h/ansible-etcd.git\r\n  version: $ENV{SOME_ENV_VARIABLE}\r\n  dst: roles/retr0h.ansible-etcd/\r\n\r\nMy specific desire is to just have this for versions, but it probably should work for all config fields if the functionality is not seen as counter to design ideas. I've already started working on this for my purposes but raised to issue to determine if a feature like this fits into the projects desire/vision.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/37", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/37/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/37/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/37/events", "html_url": "https://github.com/retr0h/gilt/issues/37", "id": 229176376, "node_id": "MDU6SXNzdWUyMjkxNzYzNzY=", "number": 37, "title": "Feature: determine what's changed in the overlay copy compared to the git source", "user": {"login": "woneill", "id": 1393558, "node_id": "MDQ6VXNlcjEzOTM1NTg=", "avatar_url": "https://avatars1.githubusercontent.com/u/1393558?v=4", "gravatar_id": "", "url": "https://api.github.com/users/woneill", "html_url": "https://github.com/woneill", "followers_url": "https://api.github.com/users/woneill/followers", "following_url": "https://api.github.com/users/woneill/following{/other_user}", "gists_url": "https://api.github.com/users/woneill/gists{/gist_id}", "starred_url": "https://api.github.com/users/woneill/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/woneill/subscriptions", "organizations_url": "https://api.github.com/users/woneill/orgs", "repos_url": "https://api.github.com/users/woneill/repos", "events_url": "https://api.github.com/users/woneill/events{/privacy}", "received_events_url": "https://api.github.com/users/woneill/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 478592830, "node_id": "MDU6TGFiZWw0Nzg1OTI4MzA=", "url": "https://api.github.com/repos/retr0h/gilt/labels/question", "name": "question", "color": "cc317c", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2017-05-16T21:36:44Z", "updated_at": "2017-06-12T22:42:06Z", "closed_at": "2017-06-12T22:41:57Z", "author_association": "NONE", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Feature request\r\n\r\n# Gilt details\r\n\r\n```\r\ngilt --version\r\ngilt, version 1.1.0\r\n```\r\n\r\n- Gilt installation method: pip\r\n\r\n# Desired Behaviour\r\n\r\nNot sure how best to implement this considering the whole point of Gilt is overlaying repos, but it'd be nice to be able to determine what's changed in the overlay copy compared to the git source.\r\n\r\nMy use case in particular is a dotfiles repo. I use Gnu Stow to create symlinks from my home folder's dotfiles to my [checked out repo](/woneill/dotfiles).\r\n\r\nW/o Gilt, I can just do a `git diff` to see what's changed. Now I'm trying to overlay a 'work' repo on top of it via Gilt but I lose that ability to easily see what's changed.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/retr0h/gilt/issues/26", "repository_url": "https://api.github.com/repos/retr0h/gilt", "labels_url": "https://api.github.com/repos/retr0h/gilt/issues/26/labels{/name}", "comments_url": "https://api.github.com/repos/retr0h/gilt/issues/26/comments", "events_url": "https://api.github.com/repos/retr0h/gilt/issues/26/events", "html_url": "https://github.com/retr0h/gilt/issues/26", "id": 214633439, "node_id": "MDU6SXNzdWUyMTQ2MzM0Mzk=", "number": 26, "title": "Replace giturlparse.py with something maintainable", "user": {"login": "beevee", "id": 1009675, "node_id": "MDQ6VXNlcjEwMDk2NzU=", "avatar_url": "https://avatars2.githubusercontent.com/u/1009675?v=4", "gravatar_id": "", "url": "https://api.github.com/users/beevee", "html_url": "https://github.com/beevee", "followers_url": "https://api.github.com/users/beevee/followers", "following_url": "https://api.github.com/users/beevee/following{/other_user}", "gists_url": "https://api.github.com/users/beevee/gists{/gist_id}", "starred_url": "https://api.github.com/users/beevee/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/beevee/subscriptions", "organizations_url": "https://api.github.com/users/beevee/orgs", "repos_url": "https://api.github.com/users/beevee/repos", "events_url": "https://api.github.com/users/beevee/events{/privacy}", "received_events_url": "https://api.github.com/users/beevee/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 3, "created_at": "2017-03-16T08:47:53Z", "updated_at": "2017-11-01T05:31:27Z", "closed_at": "2017-11-01T05:31:27Z", "author_association": "NONE", "active_lock_reason": null, "body": "# Issue Type\r\n\r\n- Feature request\r\n\r\n# Gilt details\r\n\r\n```\r\ngilt, version 1.0.0\r\n```\r\n\r\n- Gilt installation method: pip\r\n\r\n# Desired Behaviour\r\n\r\nI have a hosted GitLab instance at `git.companyname.com`. When I try to overlay a repository from this instance, I get output like this:\r\n\r\n```\r\nNone.None:\r\n  - cloning None.None to /home/user/.gilt/clone/None.None\r\n  - extracting (master) /home/user/.gilt/clone/None.None to /home/user/path/to/repository/\r\n```\r\n\r\nThat's because giturlparse.py doesn't support third-level domain names and couldn't parse my url. I can probably fix this, but the author of giturlparse.py hasn't been accepting any PRs since 2015. Worse still, tests are broken in current master branch.\r\n\r\nI propose to replace giturlparse.py with something alive and maintainable. I'm ready to do the job, if you are willing to accept such PR.", "performed_via_github_app": null, "score": 1.0}]}